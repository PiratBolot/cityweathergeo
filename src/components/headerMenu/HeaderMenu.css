.app_header {
    background-color: white;
    margin: 5px 5px 10px 5px;
    border-radius: 5px;
    box-shadow: 0 0 5px 0 #202020;
    padding: 10px;
    text-align: start;
    color: #435257;
    font-weight: bold;
    font-size: 24px;

    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}

.app_header_text {
    margin-right: 20px;
}

.app_geo_update_button {
    background-color: #a7afb1;
    color: white;
    padding: 10px;
    border: none;
    border-radius: 3px;
    font-weight: bold;
}