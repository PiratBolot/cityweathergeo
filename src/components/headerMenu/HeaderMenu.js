import React from 'react';

class HeaderMenu extends React.Component {
    render = () => (
        <div className="app_header">
            <div className="app_header_text">Погода здесь</div>
            <button className="app_geo_update_button" onClick={this.getGeoData}>Обновить геолокацию</button>
        </div>
    )
}