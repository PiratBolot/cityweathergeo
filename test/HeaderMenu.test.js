import React from 'react';
import {shallow} from '../enzyme/enzyme';
import {applyMiddleware, createStore} from "redux";
import rootReducer from "../src/reducers/Reducers";
import thunk from "redux-thunk";
import toJson from "enzyme-to-json";
import HeaderMenu from "../src/components/headerMenu/HeaderMenu";

describe('Header Menu', () => {
    const initialState = {
        trackedCities: ['Syktyvkar', 'Moscow', 'Kiev'],
        errorMessage: {isError: false, errorMsg: ""},
        geolocation: {}
    };
    let store,wrapper;
    beforeEach(()=>{
        store = createStore(rootReducer, initialState, applyMiddleware(thunk));
        wrapper = shallow(<HeaderMenu store={store} />);
    })
    it('+++capturing Snapshot of HeaderMenu', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});