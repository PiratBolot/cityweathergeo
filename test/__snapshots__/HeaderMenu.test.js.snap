// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Menu +++capturing Snapshot of HeaderMenu 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(HeaderMenu)>
    <HeaderMenu
      getLocation={[Function]}
    >
      <div
        className="app_header"
      >
        <div
          className="app_header_text"
        >
          Погода здесь
        </div>
        <button
          className="app_geo_update_button"
          onClick={[Function]}
        >
          Обновить геолокацию
        </button>
      </div>
    </HeaderMenu>
  </Connect(HeaderMenu)>
</Provider>
`;
